cdap-proto/src/main/java/co/cask/cdap/proto/codec/IdTypeAdapter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.codec;
+
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.element.ElementType;
+import co.cask.cdap.proto.id.ElementId;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+
+import java.lang.reflect.Type;
+
+/**
+ * Class for serialize/deserialize Id object to/from json through {@link com.google.gson.Gson Gson}.
+ * Uses {@link ElementId} as the actual object to be serialized and deserialized.
+ */
+public final class IdTypeAdapter extends AbstractSpecificationCodec<Id> {
```

Comment:
It's not really a specification. Why need to extend from `AbstractSpecificationCodec`?